### DISCUSSION TITLE
PuTTY.PuTTY-installed/-preuninstall

### DISCUSSION THREAD

**Discussion by KnifMelti** [Romanitho/Winget-AutoUpdate#259] [MODs]
With the latest fails of PuTTY:
There's more to `PuTTY.PuTTY` than meets the eye... https://github.com/Romanitho/Winget-AutoUpdate/discussions/249#discussioncomment-4578146
...in a **PuTTY.PuTTY-installed.ps1** you must now copy:
`C:\Windows\System32\config\systemprofile\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\PuTTY (64-bit)`
To:
`C:\ProgramData\Microsoft\Windows\Start Menu\Programs`
That's because if it installs under **System** context it places the shortcuts totally wrong...
...and when uninstalling in a **PuTTY.PuTTY-preuninstall.ps1**:
Search for **PuTTY** in the registry uninstall and Uninstall the MSI/delete the directory:
`C:\ProgramData\Microsoft\Windows\Start Menu\Programs\PuTTY (64-bit)`

All this can be accomplished by using the files **_AppID-template.ps1**/**_Mods-Functions.ps1** in my latest PR #257 (attached here)
**Winget Land**...

[PuTTY.PuTTY-installed (-preuninstall).zip](https://github.com/Romanitho/Winget-AutoUpdate/files/10369895/PuTTY.PuTTY-installed.-preuninstall.zip)

**Comment by littonztl** [comment]
How do you add the scripts in your ZIP to your WAU deployment?

**Reply by KnifMelti** [reply]
Nowadays (2 years after the previous post) **WAU** is installed by **MSI**: https://github.com/Romanitho/Winget-AutoUpdate/releases/tag/v2.1.0 taking parameters:
https://github.com/Romanitho/Winget-AutoUpdate?tab=readme-ov-file#advanced-installation
Adding **mods** to your deployment is something you must handle by yourself (deploying the **MSI** via some script and copying your own files to the **mods** folder under **INSTALLDIR**).
But **WAU** has an automated routine to get central mods from a share, local file system or an URL via the parameter **MODSPATH** that must be reachable under **SYSTEM**.

**Reply by KnifMelti** [reply]
And nowadays PuTTY.PuTTY seems to handle installation under SYSTEM context by itself, no need for a mod anymore...

**Reply by Romanitho** [reply]
You can also have a "Mods" folder in the same directory as your `wau.msi` file. Check out how it works in this script:
https://github.com/Romanitho/Winget-AutoUpdate/blob/948ef48468e845c8179dd53238ee0b24f61189e7/Sources/Winget-AutoUpdate/config/WAU-MSI_Actions.ps1#L125C1-L130C10
